src/client/fuse_ll.cc

```diff
@@ -613,9 +614,28 @@ static void fuse_ll_read(fuse_req_t req, fuse_ino_t ino, size_t size, off_t off,
   Fh *fh = reinterpret_cast<Fh*>(fi->fh);
   bufferlist bl;
   int r = cfuse->client->ll_read(fh, off, size, &bl);
-  if (r >= 0)
-    fuse_reply_buf(req, bl.c_str(), bl.length());
-  else
+  if (r >= 0) {
+    vector<iovec> iov;
+    bl.prepare_iov(&iov);
+#ifdef CEPH_HAVE_SPLICE
+    size_t len = sizeof(struct fuse_bufvec) + sizeof(struct fuse_buf) * (iov.size() - 1);
+    struct fuse_bufvec *bufv = (struct fuse_bufvec *)calloc(1, len);
+    int i = 0;
+
+    if (bufv) {
+      bufv->count = iov.size();
+      for (auto &v: iov) {
+        bufv->buf[i].mem = v.iov_base;
+        bufv->buf[i++].size = v.iov_len;
+      }
+      fuse_reply_data(req, bufv, FUSE_BUF_SPLICE_MOVE);
```

Comment:
Can we actually move the pages? The bufferlist refers to memory in the object cacher (`osdc`). I'm unsure if there is a benefit or even what `FUSE_BUF_SPLICE_MOV` is doing for us.

Reply:
It won't move the bufferlist memories, in the libfuse it will copy the data from bufferlist and write to one pipe fd, and then do the splice move from the pipe fd.